pkg/crypto/aes.go

```diff
@@ -1,51 +1,134 @@
 package crypto
 
 import (
+	"bytes"
+	"context"
 	"crypto/aes"
 	"crypto/cipher"
 	"crypto/rand"
 	"encoding/base64"
+	"fmt"
+	"os"
 
 	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-type AESCipher struct {
+type aesCipher struct {
 	key    []byte
 	cipher cipher.AEAD
 	nonce  []byte
 }
 
 const keyLength = 24 // 192 bit
 
-func NewAESCipher() (*AESCipher, error) {
+var decryptionCiphers []*aesCipher // used to decrypt data
+var encryptionCipher *aesCipher    // used to encrypt data
+
+// add cipher from API_ENCRYPTION_KEY environment variable if it is present (and set that key to be used for encryption)
+func init() {
+	decryptionCiphers = []*aesCipher{}
+	if os.Getenv("API_ENCRYPTION_KEY") != "" {
+		envCipher, err := aesCipherFromString(os.Getenv("API_ENCRYPTION_KEY"))
+		if err != nil {
+			// do nothing
+		} else {
+			decryptionCiphers = append(decryptionCiphers, envCipher)
+			encryptionCipher = envCipher
+		}
+	}
+}
+
+// InitFromSecret reads the encryption key from kubernetes and adds it to the list of decryptionCiphers, and sets this key to be used for encryption.
+func InitFromSecret(clientset kubernetes.Interface, namespace string) error {
+	sec, err := clientset.CoreV1().Secrets(namespace).Get(context.Background(), "kotsadm-encryption", metav1.GetOptions{})
```

Comment:
Is there a place where this secret is created?

Reply:
yes - https://github.com/replicatedhq/kots/blob/4cc84711e2f097984ad0d2b23dd2e2e036528e66/pkg/kotsadm/objects/secrets_objects.go#L99-L116